src/components/History/History.tsx

```diff
@@ -1,32 +1,116 @@
-import React, {
-  useEffect,
-  useState,
-} from 'react';
-
-import {
-  getItem,
-} from '../../services/storage';
+import React, { useEffect, useState } from 'react';
+
+import { getItem, setItem } from '../../services/storage';
 import { STORED_GROUPED_CROSS_CHAIN_ACTIONS } from '../../constants/storageConstants';
 import ActionPreview from '../TransactionPreview/ActionPreview';
-import { ICrossChainAction } from '../../types/crossChainAction';
+import { ICrossChainAction, ICrossChainActionTransaction } from '../../types/crossChainAction';
+
+// Hooks
+import { useEtherspot } from '../../hooks';
+
+//utils
+import { fetchTransactionsData } from '../../utils/transaction';
+import { CHAIN_ID } from '../../utils/chain';
+
+//constants
+import { TRANSACTION_BLOCK_TYPE } from '../../constants/transactionBuilderConstants';
+import { CROSS_CHAIN_ACTION_STATUS } from '../../constants/transactionDispatcherConstants';
+
+import { uniqueId } from 'lodash';
 
 const History = () => {
-  const [storedGroupedCrossChainActions, setStoredGroupedCrossChainActions] = useState<{ [id: string]: ICrossChainAction[] }>({})
+  const [storedGroupedCrossChainActions, setStoredGroupedCrossChainActions] = useState<{
+    [id: string]: ICrossChainAction[];
+  }>({});
+  const { getSdkForChainId, accountAddress, sdk } = useEtherspot();
 
   const getLatest = () => {
     try {
-      const storedGroupedCrossChainActionsRaw = getItem(STORED_GROUPED_CROSS_CHAIN_ACTIONS);
-      const storedGroupedCrossChainActionsUpdated = storedGroupedCrossChainActionsRaw
-        ? JSON.parse(storedGroupedCrossChainActionsRaw)
-        : {};
-      setStoredGroupedCrossChainActions(storedGroupedCrossChainActionsUpdated);
+      let storedGroupedCrossChainActionsUpdated;
+      let storedGroupedCrossChainActionsRaw = getItem(STORED_GROUPED_CROSS_CHAIN_ACTIONS);
+      if (storedGroupedCrossChainActionsRaw) {
+        storedGroupedCrossChainActionsUpdated = storedGroupedCrossChainActionsRaw
+          ? JSON.parse(storedGroupedCrossChainActionsRaw)
+          : {};
+        setStoredGroupedCrossChainActions(storedGroupedCrossChainActionsUpdated);
+      } else {
+      }
     } catch (e) {
       //
     }
   };
+  const getTransactionsData = async () => {
+    const sdk = getSdkForChainId(CHAIN_ID.POLYGON);
+    try {
+      if (!sdk) return;
+
+      let results = await fetchTransactionsData(sdk, accountAddress);
```

Comment:
Why are we only getting transactions for Polygon? Shouldn't we be getting history for all chains?